frontend/controller/actions/identity.js

```diff
@@ -343,6 +343,11 @@ export default (sbp('sbp/selectors/register', {
       sbp('state/vuex/commit', 'login', loginAttributes)
       await sbp('chelonia/storeSecretKeys', () => transientSecretKeys)
 
+      if (!state) {
+        // Make sure we don't unsubscribe from our own identity contract
+        await sbp('chelonia/contract/retain', identityContractID)
+      }
```

Comment:
A few questions here:

1. Could this be moved to an `else` branch of the same `if (state)` check above? (e.g. on line 329?
2. On line 368 it says someting about the sync not being awaited... but it is awaited. Is this comment something we forgot to clean up related to #1765 ? (Can it be boyscoutted now?)

Reply:
> 1. Could this be moved to an else branch of the same if (state) check above? (e.g. on line 329?

It needs to happen _after_ `await sbp('chelonia/storeSecretKeys', () => transientSecretKeys)`. If some secret keys are missing, since this is possibly calling `sync`, it could result in an invalid state.

> On line 368 it says someting about the sync not being awaited... but it is awaited.

Right.

> Is this comment something we forgot to clean up related to #1765 ? (Can it be boyscoutted now?)

I think you're right.